pkg/webhook/policy.go

```diff
@@ -480,6 +509,81 @@ func (h *validationHandler) reviewRequest(ctx context.Context, req *admission.Re
 	}
 
 	trace, dump := h.tracingLevel(ctx, req)
+	review, err := h.createReview(ctx, req)
+
+	// Check if the requested resource is a generator resource that is configured
+	// to be expanded by TemplateExpansions
+	gvk := schema.GroupVersionKind{
+		Group:   req.Kind.Group,
+		Version: req.Kind.Version,
+		Kind:    req.Kind.Kind,
+	}
+	expansionTemps := h.expansionSystem.TemplatesForGVK(gvk)
+	// If there are no matching ExpansionTemplates, review the resource normally
+	if len(expansionTemps) == 0 {
+		return h.review(ctx, review, trace, dump)
+	}
+
+	// Convert the request's generator resource to unstructured for expansion
+	obj := &unstructured.Unstructured{}
+	if _, _, err := deserializer.Decode(req.Object.Raw, nil, obj); err != nil {
+		return nil, fmt.Errorf("error decoding generator resource %s: %s", req.Name, err)
+	}
+	obj.SetNamespace(req.Namespace)
+
+	// Expand the generator and apply mutators to the resultant resources
+	resultants, err := h.expansionSystem.ExpandGenerator(obj, expansionTemps)
+	if err != nil {
+		return nil, fmt.Errorf("error expanding generator: %s", err)
+	}
+	for _, res := range resultants {
+		mutable := &mutationtypes.Mutable{
+			Object:    res,
+			Namespace: review.Namespace,
```

Comment:
here we are setting the namespace to be equivalent to the parent object's namespace, which is different from the shell command